packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -0,0 +1,139 @@
+            balance: '0', // default balance
+            availableBalance: '0', // default balance
+            empty: true,
+            history: {
+                total: -1,
+                unconfirmed: 0,
+                transactions: undefined,
+            },
+        };
+        return Promise.resolve({
+            type: RESPONSES.GET_ACCOUNT_INFO,
+            payload: account,
+        } as const);
+    }
+
+    const account: AccountInfo = {
+        descriptor: payload.descriptor,
+        balance: accountInfo.lamports.toString(), // TODO(vl): check if this should also include staking balances
+        availableBalance: accountInfo.lamports.toString(), // TODO(vl): revisit to make sure that what getAccountInfo returns is actually available balance
+        empty: accountInfo.lamports === 0, // TODO(vl): this is not correct, it should depend on the length of transaction history
+        // TODO(vl): transaction history
+        history: {
+            total: -1,
+            unconfirmed: 0,
+            transactions: undefined,
+        },
+    };
+    return Promise.resolve({
+        type: RESPONSES.GET_ACCOUNT_INFO,
+        payload: account,
+    } as const);
+};
+
+const getInfo = async (request: Request<MessageTypes.GetInfo>) => {
+    const api = await request.connect();
+    const blockHeight = await api.getBlockHeight('finalized');
+
+    // more info for the parsedBlock endpoint here https://www.quicknode.com/docs/solana/getParsedBlock
+    // the maxSupportedTransactionVersion might need to get updated if solana introduces a new version of transactions
+    const { blockhash: blockHash } = await api.getParsedBlock(blockHeight, {
+        maxSupportedTransactionVersion: 0,
+    });
+    const serverInfo = {
+        // genesisHash is reliable identifier of the network, for mainnet the genesis hash is 5eykt4UsFv8P8NJdTREpY1vzqKqZKvdpKuc147dw2N9d
+        testnet: (await api.getGenesisHash()) !== '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdpKuc147dw2N9d',
+        blockHeight,
+        blockHash,
+        shortcut: 'sol',
+        url: api.rpcEndpoint,
+        name: 'Solana',
+        version: (await api.getVersion())['solana-core'],
+        decimals: 9,
+    };
+    return {
+        type: RESPONSES.GET_INFO,
+        payload: { ...serverInfo },
+    } as const;
+};
+
+const onRequest = (request: Request<MessageTypes.Message>) => {
+    switch (request.type) {
+        case MESSAGES.GET_ACCOUNT_INFO:
+            return getAccountInfo(request);
+        case MESSAGES.GET_INFO:
+            return getInfo(request);
+        default:
+            throw new CustomError('worker_unknown_request', `+${request.type}`);
+    }
+};
+
+class SolanaWorker extends BaseWorker<SolanaAPI> {
+    protected isConnected(api: SolanaAPI | undefined): api is SolanaAPI {
+        return !!api;
+    }
+
+    tryConnect(url: string): Promise<SolanaAPI> {
+        const api = new Connection(`https://${url}`, { wsEndpoint: `wss://${url}` });
```

Comment:
Solana is a bit similar to Ripple in this case, there is no `websocket` file which implements the API, `Connection` already is that implementation, it uses https for requests and websocket for subscribe methods like `onAccountChange`.